packages/core/services/signers/DashMemoryDevice.js

```diff
@@ -3,13 +3,22 @@
  * Licensed under the AGPL Version 3 license.
  */
 
-import EventEmitter from 'events'
+import bitcoin from 'bitcoinjs-lib'
+import { Address, Networks, PrivateKey, PublicKey } from 'dashcore-lib'
 
-export default class DashMemoryDevice extends EventEmitter {
-  constructor () {
-    super()
-    Object.freeze(this)
+export default class DashMemoryDevice {
+  constructor ({ privateKey, network }) {
+    this.privateKey = privateKey;
+    this.network = network;
+    Object.freeze(this);
   }
 
-  // @todo going to be implemented
+  getAddress () {
+    const networkType = this.network === bitcoin.networks.testnet ? Networks.testnet : Networks.livenet;
```

Comment:
As far as I know `bitcoin.networks.testnet` is object: https://github.com/bitcoinjs/bitcoinjs-lib/blob/master/src/networks.js

Is this condition correct?

Reply:
I took the approach from https://github.com/ChronoBank/ChronoMint/blob/develop/packages/core/redux/bitcoin/utils.js#L75. It works actually. In testnet mode a test address (starts with 'y') is generated and requests are submitted to the testnet node (testnet-insight.dashevo.org). In mainnet mode a mainnet address is generated (begins with 'X') is generated and requests are submitted to the mainnet node (insight.dashevo.org).

Reply:
Hmm. I do not understand how it may work. We are comparing string and object.
Also I'm not sure that approach from  https://github.com/ChronoBank/ChronoMint/blob/develop/packages/core/redux/bitcoin/utils.js#L75 is also correct.